### DISCUSSION TITLE
Can we control disk size for codespaces?

### DISCUSSION THREAD

**Discussion by ghost** [devcontainers/community#8] [Q&A]
Can we define the disk size in the devcontainer.json file? The default setting seems to be a 32gb disk, but that burns through the 15.00GB of monthly storage too fast and is far more disk space than I need for a dedicated but occasionally used remote dev environment.

**Comment by ghost** [comment]
I tried to add a `hostRequirements`  [here](https://github.com/Pavocracy/dotfiles/blob/main/.devcontainer/devcontainer.json#L23) but it didn't seem to limit the disk space

**Reply by Chuxel** [reply]
There is not a setting like that today in the dev continaer spec, though that could make sense to add.

That said, and as far as I know, the disk size is tied to the compute size in Codespaces. That said, I believe limit is based on consumed space, not potential space, but you can find more information in the Codespaces specific discussions.

https://github.com/community/community/discussions/categories/codespaces

e.g. 
- https://github.com/community/community/discussions/39697
- https://github.com/community/community/discussions/38697
